src/core/libmaven/csvreports.cpp

```diff
@@ -131,112 +131,62 @@ void CSVReports::insertPeakInformationIntoCSVFile(PeakGroup* group) {
 
 }
 
-void CSVReports::insertGroupInformationIntoCSVFile (PeakGroup* group) {
-
+void CSVReports::insertGroupInformationIntoCSVFile(PeakGroup* group)
+{
     if(group->compound == NULL || group->childCount() == 0) {
-
         writeGroupInfo(group);
-
-    }
-
-    else {
-
-        string formula = group->compound->formula;
-        int charge = getMavenParameters()->getCharge(group->compound);
-        bool C13Flag = getMavenParameters()->C13Labeled_BPE;
-        bool N15Flag = getMavenParameters()->N15Labeled_BPE;
-        bool S34Flag = getMavenParameters()->S34Labeled_BPE;
-        bool D2Flag = getMavenParameters()->D2Labeled_BPE;
-
-        vector<Isotope> masslist = MassCalculator::computeIsotopes(
-            formula,
-            charge,
-            C13Flag,
-            N15Flag,
-            S34Flag,
-            D2Flag
-        );
-        insertIsotopes(group,masslist);
-
+    } else {
+        insertIsotopes(group);
     }
 }
 
-int CSVReports::getIonisationMode() {
-
-      int ionizationMode;
-
-      if (samples.size() > 0 && samples[0]->getPolarity() > 0)
-          ionizationMode = +1;
-      else
-          ionizationMode = -1;
-
-      return ionizationMode;
-}
-
-void CSVReports::insertIsotopes (PeakGroup* group, vector<Isotope> masslist) {
-
-      for (unsigned int i = 0; i < masslist.size(); i++) {
-                Isotope& x = masslist[i];
-                string isotopeName = x.name;
-                insertUserSelectedIsotopes(group,isotopeName);
-      }
-}
-
-void CSVReports::insertUserSelectedIsotopes(PeakGroup* group, string isotopeName) {
-
-      int counter = 0;
-      for (unsigned int k = 0; k < group->children.size() && counter == 0; k++) {
-          //output non-zero-intensity peaks
-          counter = insertIsotpesFoundInSamples(group, isotopeName, counter, k);
-      }
-      /*
-       * Commented out in Maven776 - Kiran
-      if (counter == 0) {
-          //output zero-intensity peaks
-          insertIsotpesNotFoundInSamples(group, isotopeName);
-      }
-      */
+void CSVReports::insertIsotopes(PeakGroup* group, bool userSelectedIsotopesOnly)
+{
+    if (userSelectedIsotopesOnly) {
+        insertUserSelectedIsotopes(group);
```

Comment:
The default value of `userSelectedIsotopesOnly` is **false** and no where in the code I see it being called as **true**.
And if that's the case, `insertUserSelectedIsotopes(group)` should not even exist.
Is this the case or did I miss something ?

Reply:
From #882, it seems in future we might ask users to choose which labels to export.
If that's the reason, can we comment out the code related to this ? and enable it once we have the functionality. Won't be compiled unnecessarily
Not sure what to do with this
@shubhra-agrawal  
What do you think ?

Reply:
So the changes made will by default export all the isotope labels that were detected for a group. @shubhra-agrawal suggested that there should still be an option to allow the user to export only those labels which they have selected in the "Options->Isotope Detection" settings (and filter the others out, even if they were detected), as it has been doing until now. Providing that function is the purpose of this method. Since most of the code for this method was already present, I am just refactoring and leaving it here until we have a UI allowing us to expose this on the front-end (hopefully with the UX overhaul we will have this as one of the options in our export tab).